src/test/java/it/pagopa/pn/cucumber/steps/pg/LegalPersonVirtualKeySteps.java

```diff
@@ -1,69 +1,360 @@
             }
+            case "pg" -> {
+                virtualKeyServiceClient.setBearerToken(SettableBearerToken.BearerTokenType.PG_5);
+                tosPrivacyClient.setBearerToken(SettableBearerToken.BearerTokenType.PG_5);
+            }
+            case "amministratore di un altra pg" -> {
+                virtualKeyServiceClient.setBearerToken(SettableBearerToken.BearerTokenType.PG_2);
+                tosPrivacyClient.setBearerToken(SettableBearerToken.BearerTokenType.PG_2);
+            }
+            default -> throw new IllegalArgumentException("ADMIN NOT VALID");
+        }
+    }
+
+    private class VirtualKeyExpectedResponse {
+        BffNewVirtualKeyResponse response;
+        @Setter
+        VirtualKeyState state;
+        String user;
+
+        private VirtualKeyExpectedResponse(BffNewVirtualKeyResponse response, VirtualKeyState state, String user) {
+            this.response = response;
+            this.state = state;
+            this.user = user;
+        }
+
+        private VirtualKeyExpectedResponse(BffNewVirtualKeyResponse response) {
+            this.response = response;
+        }
+    }
+
+    @After("@removeAllVirtualKeyTEst")
+    public void removeVirtualKey() {
+        //se la chiave pubblica è cancellata devo riattivarla per eliminare le virtual key e poi ricancellarla.
+        selectPGUser("amministratore");
+        BffVirtualKeysResponse getVirtualKeys = virtualKeyServiceClient.getVirtualKeys(null, null, null, null);
+        getVirtualKeys.getItems().stream()
+            .filter(data -> data.getStatus() != null && data.getStatus().getValue().equals(VirtualKeyState.BLOCKED.getState()))
+            .findFirst()
+            .ifPresent(data -> virtualKeyCancellation(data.getId()));
+
+        getVirtualKeys.getItems().forEach(data -> {
+            if (data.getStatus().getValue().equals(VirtualKeyState.ENABLE.getState()) || data.getStatus().getValue().equals(VirtualKeyState.REACTIVE.getState())) {
+                BffVirtualKeyStatusRequest requestNewVirtualKey = new BffVirtualKeyStatusRequest();
+                requestNewVirtualKey.setStatus(BffVirtualKeyStatusRequest.StatusEnum.BLOCK);
+                Assertions.assertDoesNotThrow(() -> virtualKeyServiceClient.changeStatusVirtualKeys(data.getId(), requestNewVirtualKey));
+                Assertions.assertDoesNotThrow(() -> virtualKeyCancellation(data.getId()));
+            } else if (!data.getStatus().getValue().equals(VirtualKeyState.BLOCKED.getState())){
+                Assertions.assertDoesNotThrow(() -> virtualKeyCancellation(data.getId()));
+            }
+        });
+    }
+
+    @After("@removeAllVirtualKey")
+    public void removeVirtualKeyTest() {
+        String user = "AMMINISTRATORE";
+        selectPGUser(user);
+        lUtenteAccettaITos(user, ACCEPT_TOS);
+        //se la chiave pubblica è cancellata devo riattivarla per eliminare le virtual key e poi ricancellarla (faccio partire il metodo after delle public dopo).
```

Comment:
Valutare se questo commento potrebbe tornare utile, nel caso è possibile riformularlo in modo più formale?